### ISSUE TITLE
libfaac

### ISSUE THREAD

**Issue by onon765trb** [DigitalMediaServer/DigitalMediaServer#99]
There is
`							acodec += "libfaac";`
l. 1240 on MEncoderVideo.java.
Is this ok?
ffmpeg git [lavc: remove libfaac wrapper](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=dc0f711459e0c682bf9f94ba38d26736e90cff45)

**Comment by Nadahar** [comment]
MEncoder is a bit of a mess. We haven't done any changes to the MEncoder code in DMS yet, so it's exactly like it is in UMS. @SubJunk has built his own versions of MEncoder, but only for Windows as far as I know. I've tried to find what is different with that build so I could try to build it for macOS and Linux too, but I haven't found it. I don't even know exactly what is different with this version.

I therefore suspect that MEncoder is only working properly for Windows as it is, and there's probably a good deal of cleanup needed to make this work properly for all platforms. @SubJunk is currently working with a somewhat related PR https://github.com/UniversalMediaServer/UniversalMediaServer/pull/1601, but I haven't really looked into it so I don't know what it solves.

**Comment by Sami32** [comment]
@onon765trb No.
The transcoding code in general is a mess and need to be refactorized.
The MEncoder case is worst. I plan to replace it, but no ETA as far more work need to be done before.

**Comment by SubJunk** [comment]
I would be happy to release my MEncoder code to you @Nadahar 
I went far down the rabbit hole with it by also patching some of the dependencies so I would need to include those, and it was even picky about which version of GCC I used so the environment matters too. In some cases I submitted the patches to the dependencies so they were integrated but in other cases I either didn't do that, or the patches weren't merged.

Anyway I still have all the backups so feel free to try, it was quite heavily customized. A lot of the customizations were for DVD support but also audio sync fixes

**Comment by onon765trb** [comment]
I minded a licence.

**Comment by Nadahar** [comment]
@SubJunk It would be nice to have available somewhere when I have the time to look at it again some time in the future. It won't be anytime soon though, as I have too many other things at the moment.

**Comment by Nadahar** [comment]
@onon765trb 
> I minded a licence.

I don't think there's a problem with the license for us. `libfaac` seems to be GPL in older versions and LGPL in more recent versions, neither should be a problem for us (DMS itself is GPL). As far as I understand, the one with the troublesome license is `libfdk-aac` or `x264` however you choose to look at it. The point, as I understand it, is that you can't distribute something built with both these. It's fully legal for people to compile and use them together themselves though, but I don't think that's a realistic option for most users.

It seems to me that the reason FFmpeg removed `libfaac` was because they don't want to be restricted by GPL, and because `libfaac` doesn't do all that they want. I don't have enough knowledge about the details to know if there's another viable AAC encoder for MEncoder, so for now I think this is fine.

**Comment by Sami32** [comment]
>You cannot legally distribute MEncoder binaries with 
libfaac support: MEncoder is GPL and you cannot fulfill 
the requirements of the GPL for libfaac which is 
proprietary, non-free software.

Taken from a FFmpeg developper:
http://lists.mplayerhq.hu/pipermail/mencoder-users/2012-December/013268.html

**Comment by Nadahar** [comment]
OK, I just looked at the license for https://github.com/knik0/faac

It seems that it's more complicated, as they use a seemingly impossible combination of freeware and GPL for their license according to the readme.

In any case, it's not the the license of libfaac that is breached in this situation, it's MEncoder's GPL license. It seems very unlikely to me that this is the intention of the MEncoder developers since they themselves have implemented support for libfaac. Whether this part of GPL is even valid (which forbids you to depend on libraries that aren't themselves GPL) is contested, and many claim that if this had been tested in court, it would have been ruled invalid.

I'm not expert on this, but it seems to me like this is just the result of stupid licensing rules and misunderstandings rather than an actual intent of making the combination impermissible.

**Comment by Sami32** [comment]
https://www.spirton.com/mplayer-mencoder-subjunk-build/comment-page-4/#comment-38744 ;)

**Comment by Sami32** [comment]
@Onon765trb My conclusion will be that it should be added as a separate library.
